pkg/products/mcg/prometheusRules.go

```diff
@@ -0,0 +1,135 @@
+package mcg
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *Reconciler) newAlertReconciler(ctx context.Context, client k8sclient.Client, logger l.Logger, installType string) (resources.AlertReconciler, error) {
+	installationName := resources.InstallationNames[installType]
+
+	observabilityConfig, err := r.ConfigManager.ReadObservability()
+	if err != nil {
+		logger.Warning("failed to get observability config")
+		return nil, err
+	}
+
+	namespace := observabilityConfig.GetNamespace()
+	return &resources.AlertReconcilerImpl{
+		Installation: r.installation,
+		Log:          logger,
+		ProductName:  "mcg",
+		Alerts: []resources.AlertConfiguration{
+			{
+				AlertName: "mcg-operator-ksm-endpoint-alerts",
+				GroupName: "mcg-operator-endpoint.rules",
+				Namespace: namespace,
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "RHOAMMCGOperatorMetricsServiceEndpointDown",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlEndpointAvailableAlert,
+							"message": fmt.Sprintf("No {{  $labels.endpoint  }} endpoints in namespace %s. Expected at least 1.", r.Config.GetOperatorNamespace()),
+						},
+						Expr:   intstr.FromString(fmt.Sprintf("kube_endpoint_address_available{namespace='%s', endpoint='noobaa-operator-service'} < 1", r.Config.GetOperatorNamespace())),
```

Comment:
It looks like it does exist, although doesn't work as I would've expected. Instead of ready moving from true to false the metric is absent once not available. I could change it for MCG to use an absent check on it but I don't think it's wise to change all the other uses as part of this PR